packages/react-native-codegen/src/parsers/error-utils.js

```diff
@@ -265,6 +266,40 @@ function throwIfUnsupportedFunctionParamTypeAnnotationParserError(
   );
 }
 
+const UnsupportedArrayElementTypeAnnotationTypeToInvalidArrayElementTypeAnnotationTypeMap =
+  {
+    FunctionTypeAnnotation: 'FunctionTypeAnnotation',
+    VoidTypeAnnotation: 'void',
+    PromiseTypeAnnotation: 'Promise',
+  };
+
+function throwIfArrayElementTypeAnnotationIsUnsupported(
+  hasteModuleName: string,
+  flowElementType: $FlowFixMe,
+  flowArrayType: 'Array' | '$ReadOnlyArray' | 'ReadonlyArray',
+  type: string,
+  language: ParserType,
+) {
+  if (
+    type === 'VoidTypeAnnotation' ||
+    type === 'PromiseTypeAnnotation' ||
+    type === 'FunctionTypeAnnotation'
```

Comment:
How about
```
type in UnsupportedArrayElementTypeAnnotationTypeToInvalidArrayElementTypeAnnotationTypeMap
```
or maybe even look it up in the map right away and check the result vs undefined?..

Reply:
Also, that map definition could be probably inlined, in order to avoid the unwieldy name (it's only used inside this function, anyway).